### PR TITLE
Xorvoid fix timeout 0

### PR DESCRIPTION
**Pull request by jbendes**
Moved timeouts to unsigneds

### CODE REVIEW

**Comment by jbendes** [comment]
@xorvoid good to merge?

File: zcm/transport/transport_file.cpp [hunk 1]
```diff
@@ -115,15 +115,15 @@ struct ZCM_TRANS_CLASSNAME : public zcm_trans_t
...
+    int recvmsg(zcm_msg_t *msg, unsigned timeout)
     {
         assert(mode == "r");
         if (!good()) {
             // TODO Build in a way for a transport to tell zcm that an "error"
             //      has occurred. Not sure what to do here since this function
             //      has no way of communicating to the caller that this function
             //      shouldn't be called anymore
-            usleep(timeout);
+            if (timeout > 0) usleep(timeout);
```
Comment (xorvoid) [inline zcm/transport/transport_file.cpp:126]: Found another bug:

timeout is in millis 
usleep() is in micros

**Review by xorvoid — COMMENTED** [review]
Found a bug

File: zcm/transport/transport_file.cpp (hunk 1 above)
Comment (jbendes) [reply zcm/transport/transport_file.cpp:126]: yeah good catch. will fix

**Pull request merged**